xla/stream_executor/stream_executor_interface.h

```diff
@@ -359,11 +359,51 @@ class StreamExecutorInterface {
   // Returns a stream allocated by this executor, or nullptr if not found.
   virtual Stream* FindAllocatedStream(void* device_stream) { return nullptr; }
 
+  // The following methods access an internal log of some subset
+  // of arguments passed to other class methods.
+  // Used for testing/debugging purposes.
+
+  struct GemmCallTrace {
+    enum class GemmType {
+      kPlain = 0,
+      kStridedBatched = 1,
+      kBatched = 2,
+      kBlasLt = 3
+    };
+    GemmType op;
+    int flags;
+    uint64_t size1, size2;
+  };
+  // This may be expanded as necessary to trace other calls
+  using ApiTrace = std::variant<GemmCallTrace>;
+
+  // Retrieves and clears internal argument logs.
+  virtual absl::StatusOr<std::vector<ApiTrace> > ExtractApiTrace() {
+    return absl::UnimplementedError("Not implemented");
+  }
+  virtual absl::Status RecordApiTrace(ApiTrace call) {
+    return absl::UnimplementedError("Not implemented");
+  }
+
+  // Bit mask
+  enum class ArgumentLogging: uint64_t {
+    kNone = 0,
+    kGemm = 1,
+  };
+
+  // Sets the argument logging mode. Returns true if 'mode' is valid.
+  virtual bool SetArgumentLoggingMode(ArgumentLogging mode) { return false; }
+
  private:
   StreamExecutorInterface(const StreamExecutorInterface&) = delete;
   void operator=(const StreamExecutorInterface&) = delete;
 };
 
+inline bool operator&(StreamExecutorInterface::ArgumentLogging a, 
```

Comment:
This needs a comment for why it's necessary.  In general, it's problematic to bit-wise AND enum values together, as that creates undefined enum values, which is antithetical to making an enum class in the first place.

Reply:
What class/type should I use instead of enum? The intent is to allow logging to be enabled/disabled independently for individual submodules within StreamExecutor, with one flag per submodule.

Reply:
If you want a bitmask, I think a uint64_t is the most appropriate thing, and probably a bunch of constexpr uint64_t for individual masks.  

I don't think it makes sense to use "enum class" for things that are bitmasks, because you can't really AND them together and produce valid enum class values unless you get super verbose (kFoo = 1, kBar = 2, kFooBar = 3, etc), which becomes basically unreadable IMO.